common/static/common/js/discussion/views/discussion_thread_list_view.js

```diff
@@ -462,7 +462,7 @@
             DiscussionThreadListView.prototype.performSearch = function($searchInput) {
                 // trigger this event so the breadcrumbs can update as well
                 this.trigger('search:initiated');
-                this.searchFor($searchInput.val(), $searchInput);
+                this.searchForUser($searchInput.val(), $searchInput);
```

Comment:
Good question:
`searchFor ` method search the data related to the text. For example, you have search "example" it returns the data related to the text. but if you search the user so it only returns the username notifications but it returns the data based on text search. 
That why I have changed the method `searchFor` to `searchForUser.` if user's discussions exist it returns the data else returns the data based on text search. Make sense?